NSS-Day-Cohort-69/tabloid-one-more-div
# Feature - Delete Posts

### Purpose
To allow Users to delete their own Posts, and allow Admins to delete unapproved Posts

### Files Changed
- Added DELETE endpoint that deletes a Post in `PostController.cs`
- Added fetch that deletes a Post in `postManager.js`
- Added rendering and logic for deleting a Post in `PostDetails.jsx` and `ApprovePost.jsx`

### To Test
Fetch, setup, and run according to project README, then confirm the following
- Clicking the "Delete" button on a Post in the Post Details or Unapproved Posts views causes a confirmation dialogue box to appear
- Clicking "Cancel" on the dialogue box makes the box go away, and the Post is not deleted
- Clicking "Confirm" in the dialogue box in the Post Details view deletes the Post and navigates the User to the Posts List view
- Clicking "Confirm" in the dialogue box in the Unapproved Posts view deletes the Post and rerenders the page

closes #40
